File: 883-car-fleet/car-fleet.cpp
class Solution {
public:
    int carFleet(int target, vector<int>& pos, vector<int>& speed) {
        vector<pair<int, int>> temp;
        for(int i=0;i<pos.size();i++){
            temp.push_back({pos[i], speed[i]});
        }
        sort(temp.rbegin(), temp.rend());
        vector<double> stack;
        for(auto p: temp){
            stack.push_back((double)(target-p.first)/p.second);
            if(stack.size()>=2 && stack.back()<=stack[stack.size()-2]){
                stack.pop_back();
            }
        }
        return stack.size();
    }
};

File: 883-car-fleet/car-fleet.py
class Solution:
    def carFleet(self, target: int, pos: List[int], speed: List[int]) -> int:
        pair = [(p,s) for p,s in zip(pos, speed)]
        pair.sort(reverse=True)
        fleets=1
        prevtime = (target-pair[0][0]) / pair[0][1]
        for i in range(1, len(pair)):
            curtime = (target-pair[i][0]) / pair[i][1]
            if curtime > prevtime:
                fleets += 1
                prevtime = curtime

        return fleets